### ISSUE TITLE
Bug: No name? - EntityCreate

### ISSUE THREAD

**Issue by Jakllp** [Jakllp/MyPet#63]
Reported for 1.16.5
```
[18:31:25] [Server thread/ERROR]: Could not pass event CreatureSpawnEvent to PlotSquared v6.0.10-SNAPSHOT
org.bukkit.event.EventException: null
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:310) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at org.bukkit.craftbukkit.v1_16_R3.event.CraftEventFactory.callCreatureSpawnEvent(CraftEventFactory.java:651) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at org.bukkit.craftbukkit.v1_16_R3.event.CraftEventFactory.doEntityAddEventCalling(CraftEventFactory.java:585) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at net.minecraft.server.v1_16_R3.WorldServer.addEntity0(WorldServer.java:1027) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at net.minecraft.server.v1_16_R3.WorldServer.addEntity(WorldServer.java:957) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at de.Keyle.MyPet.compat.v1_16_R3.entity.EntityRegistry.spawnMinecraftEntity(EntityRegistry.java:111) ~[?:?]
	at de.Keyle.MyPet.entity.MyPet.createEntity(MyPet.java:546) ~[?:?]
	at de.Keyle.MyPet.commands.CommandSwitch$1$1.callback(CommandSwitch.java:104) ~[?:?]
	at de.Keyle.MyPet.commands.CommandSwitch$1$1.callback(CommandSwitch.java:94) ~[?:?]
	at de.Keyle.MyPet.gui.selectionmenu.MyPetSelectionGui.lambda$open$0(MyPetSelectionGui.java:116) ~[?:?]
	at de.Keyle.MyPet.api.gui.IconMenu.on(IconMenu.java:146) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:567) ~[?:?]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at net.minecraft.server.v1_16_R3.PlayerConnection.a(PlayerConnection.java:2436) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at net.minecraft.server.v1_16_R3.PacketPlayInWindowClick.a(SourceFile:32) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at net.minecraft.server.v1_16_R3.PacketPlayInWindowClick.a(SourceFile:10) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at net.minecraft.server.v1_16_R3.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:28) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at net.minecraft.server.v1_16_R3.TickTask.run(SourceFile:18) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeTask(SourceFile:144) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at net.minecraft.server.v1_16_R3.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeNext(SourceFile:118) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at net.minecraft.server.v1_16_R3.MinecraftServer.bb(MinecraftServer.java:1061) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at net.minecraft.server.v1_16_R3.MinecraftServer.executeNext(MinecraftServer.java:1054) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.awaitTasks(SourceFile:127) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at net.minecraft.server.v1_16_R3.MinecraftServer.sleepForTick(MinecraftServer.java:1038) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:970) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$0(MinecraftServer.java:273) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	at java.lang.Thread.run(Thread.java:831) [?:?]
Caused by: java.lang.NullPointerException: Cannot invoke "String.toLowerCase(java.util.Locale)" because the return value of "org.bukkit.entity.EntityType.getName()" is null
	at com.fastasyncworldedit.bukkit.adapter.IBukkitAdapter.adapt(IBukkitAdapter.java:346) ~[?:?]
	at com.fastasyncworldedit.bukkit.adapter.IDelegateBukkitImplAdapter.adapt(IDelegateBukkitImplAdapter.java:271) ~[?:?]
	at com.sk89q.worldedit.bukkit.BukkitAdapter.adapt(BukkitAdapter.java:395) ~[?:?]
	at com.plotsquared.bukkit.util.BukkitEntityUtil.checkEntity(BukkitEntityUtil.java:348) ~[?:?]
	at com.plotsquared.bukkit.listener.EntityEventListener.creatureSpawnEvent(EntityEventListener.java:209) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:567) ~[?:?]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot-1.16.5.jar:3096-Spigot-9fb885e-296df56]
	... 36 more
```

A closer look
[ticket.zip](https://github.com/Jakllp/MyPet/files/7198922/ticket.zip)

**Comment by Jakllp** [comment]
Found this https://github.com/MyPetORG/MyPet/issues/1468

**Comment by Jakllp** [comment]
Another one
[ticket.zip](https://github.com/Jakllp/MyPet/files/7475005/ticket.zip)

**Comment by Jakllp** [comment]
Okay so here are some new findings:
- Whenever an other plugin wants to know the EntityType we return "UNKNOWN" (https://github.com/Jakllp/MyPet/blob/master/modules/v1_17_R1/src/main/java/de/Keyle/MyPet/compat/v1_17_R1/entity/CraftMyPet.java#L110)
- This will return null when .getName() is used

Fix: Return a different Type? Maybe according to the Pet we are using?

**Comment by Jakllp** [comment]
So as this isn't Java 8 anymore that isn't the solution

**Comment by Jakllp** [comment]
Should be fixed with `3.12-SNAPSHOT-B1674`
See https://github.com/MyPetORG/MyPet/commit/ab6759f05c1f802b784c7164e4c5cb4e68c28e8e
